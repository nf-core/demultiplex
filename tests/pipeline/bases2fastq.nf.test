nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "bases2fastq"
    tag "pipeline"

    test("Bases2Fastq") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/b2fq-samplesheet.csv'
                demultiplexer = 'bases2fastq'
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 7 },
                { assert snapshot(
                    // FIXME
                    // path("$outputDir/sim-data/DefaultSample_R1.fastq.gz.md5"),
                    // path("$outputDir/sim-data/DefaultSample_R2.fastq.gz.md5"),
                    path("$outputDir/sim-data/Metrics.csv"),
                    path("$outputDir/sim-data/UnassignedSequences.csv"),
                    path("$outputDir/sim-data/Samples/DefaultSample/DefaultSample_R1.fastq.gz"),
                    path("$outputDir/sim-data/Samples/DefaultSample/DefaultSample_R2.fastq.gz"),
                ).match("bases2fastq") },
                { assert new File("$outputDir/sim-data/Bases2Fastq-Sim_QC.html").exists() },
                { assert new File("$outputDir/sim-data/RunStats.json").exists() },
                { assert new File("$outputDir/sim-data/Samples/DefaultSample/DefaultSample_stats.json").exists() },
                { assert new File("$outputDir/sim-data/RunManifest.json").exists() },
            )
        }

    }

}
