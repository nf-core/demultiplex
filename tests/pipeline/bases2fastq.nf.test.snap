{
    "software_versions": {
        "content": [
            "{BASES2FASTQ={bases2fastq=1.1.0.593880262, use subject to license available at elementbiosciences.com}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, UNTAR={untar=1.30}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T07:59:51+0000"
    },
    "bases2fastq": {
        "content": [
            "Metrics.csv:md5,87702e562aa1091e70e5d6e2e19f10b0",
            "RunManifest.json:md5,d4ecdb19e5fbcffab3e6e770c34023fb",
            "UnassignedSequences.csv:md5,11c1693830ce941b8cfb8d2431a59097",
            "DefaultSample_R1.fastq.gz:md5,5b631c9bec4fe8227a467bfcc382654c",
            "DefaultSample_R2.fastq.gz:md5,7318b86370fdbc87f989c45c73331e38"
        ],
        "timestamp": "2023-10-17T07:59:51+0000"
    }
}